enzyme/Enzyme/InstructionDerivatives.td

```diff
@@ -839,6 +892,16 @@ def : IntrPattern<(Op $x, $y),
                   (Select (FCmpOLT $x, $y), (SelectIfActive $y, (Shadow $y), (Zero $y)), (SelectIfActive $x, (Shadow $x), (Zero $x)))
                   >;
 
+def : CallPattern<(Op $x, $y),
+                  ["fdim", "fdimf", "fdiml"],
+                  [
+                    (Select (FCmpOLT $x, $y), (ConstantFP<"0"> $x), (DiffeRet)),
```

Comment:
This feels wrong?

if fdim(x, y) = abs|x-y| then shouldnt this be either differet or -differet?

Reply:
[`fdim`](https://en.cppreference.com/w/cpp/numeric/math/fdim) is max(x-y,0), except for corner cases related to NaN.